core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -300,8 +300,16 @@ private[spark] class Executor(
               m
             }
           }
-          val taskEndReason = new ExceptionFailure(t, metrics)
-          execBackend.statusUpdate(taskId, TaskState.FAILED, ser.serialize(taskEndReason))
+          val serializedTaskEndReason = {
+            try {
+              ser.serialize(new ExceptionFailure(t, metrics))
```

Comment:
(By the way, the particular exception I'm thinking of is Scala's MatchError, which actually includes the object which failed to match.)